### PR TITLE
Use new reactor contextWrite when available (from reactor 3.4.0)

### PR DESCRIPTION
**Pull request by trask**
Related to #7107 and #7202

Support WebFlux 6. 

Supporting reactor 3.5 seems pretty straightforward, the `subscriberContext()` was deprecated in 3.4 in favor of `contextWrite()`. In 3.5, `subscriberContext()` was removed.

This PR doesn't bump `latestDepTestLibrary` to 3.5 yet because there are a couple of tests that succeed in 3.4 using `contextWrite()` but fail in 3.5 using `contextWrite()`.

My proposal is to review/merge this PR, and then I can ping our resident reactor experts to see if they have thoughts on the failing tests in 3.5.

### CODE REVIEW

File: instrumentation/spring/spring-webflux-5.0/library/src/main/java/io/opentelemetry/instrumentation/spring/webflux/client/SpringWebfluxHttpAttributesGetter.java [hunk 1]
```diff
@@ -56,6 +67,7 @@ public List<String> requestHeader(ClientRequest request, String name) {
   }
 
   @Override
+  @Nullable
   public Integer statusCode(
```
Comment (mateuszrzeszutek) [inline instrumentation/spring/spring-webflux-5.0/library/src/main/java/io/opentelemetry/instrumentation/spring/webflux/client/SpringWebfluxHttpAttributesGetter.java:71]: I encountered the same issue in #7551 -- and decided not to use the `rawStatusCode()` method, it's already deprecated and a target for removal.

File: instrumentation/reactor/reactor-3.1/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/reactor/ContextPropagationOperatorInstrumentationTest.java [hunk 2]
```diff
@@ -136,18 +160,25 @@ void storeContextForcesItToBecomeCurrent() {
...
+                            // context
+                            // written in the next call
+                            return new ExtensionAnnotationsTracedWithSpan()
+                                .mono(Mono.just("Value"));
+                          });
+              try {
+                interim =
+                    (Mono<String>)
+                        MONO_CONTEXT_WRITE_METHOD.invoke(
+                            interim, new StoreOpenTelemetryContext(Context.current().with(span)));
```
Comment (mateuszrzeszutek) [inline instrumentation/reactor/reactor-3.1/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/reactor/ContextPropagationOperatorInstrumentationTest.java:177]: In the previous version of this code, wasn't `Context.current().with(span)` executed as a part of the lambda? I mean it was lazy before, now it's eager

**Review by mateuszrzeszutek — APPROVED** [review]

**Comment by mateuszrzeszutek** [comment]
> This PR doesn't bump `latestDepTestLibrary` to 3.5 yet because there are a couple of tests that succeed in 3.4 using `contextWrite()` but fail in 3.5 using `contextWrite()`.

That's the part where I gave up when I was trying to update the instrumentation 🙈 
Still, I had no idea that Webflux did not really need it -- good for us!

File: instrumentation/spring/spring-webflux-5.0/library/src/main/java/io/opentelemetry/instrumentation/spring/webflux/client/SpringWebfluxHttpAttributesGetter.java (hunk 1 above)
Comment (trask) [reply instrumentation/spring/spring-webflux-5.0/library/src/main/java/io/opentelemetry/instrumentation/spring/webflux/client/SpringWebfluxHttpAttributesGetter.java:43]: I kept `rawStatusCode()` for version range `[5.1,6.0)`, but added new path for 6.0

**Pull request merged**